README.md
# AESSET Resource Manager
test

Some changes to test how things work
And some more changes

Changes for 0.1.4 Xxxxxxxxxxxxxxxxxxxxxxxx
We zijn lekker bezig voor 1.1.0!

TEST SSH !:wq
TEST SSH vnauit IDE

## Installation instructions
For local deployment you can run it in 2 ways: 
- in Docker on your pc
- local on your pc

The preferred way is to deploy it in Docker, since all dependencies (python, 
libraries and postgres database) are included in the Docker images. 
Only debugging and accessing the database is harder.

### Installation in Docker on your desktop
Global steps (to be refined later)
1) install Docker, see <a href="https://www.docker.com/products/docker-desktop" target="_blank">docker.com</a>. 

    After installation:

    1.1 make sure that you have a 'docker' group and that your user account 
    has been added to it. If not, create the group and add your user. On linux:
    ```
    sudo groupadd docker
    sudo gpasswd -a ${USER} docker
    ```
    For mac, see here how to setup users and groups: 
    <a href="https://support.apple.com/guide/mac-help/set-up-other-users-on-your-mac-mtusr001/mac" 
    target="_blank">support.apple.com</a>
    
    1.2 and (re)start docker

2) install git, see <a href="https://git-scm.com/downloads" target="_blank">git-scm.com</a>
   
3) checkout the code

    On the command line, go to the location where the project directory has to be
    placed, e.g. /python/django. Clone the repository which will create a subdirectory
    'stripped' in there. Type:
    ```
    git clone https://github.com/EelcoA/stripped.git
    ```
4) go into the newly created directory:
    ```
    cd stripped
    ```
5) start app:
    ```
    docker-compose up --build 
    ```
    This first time it will take a while since the base images for Python and 
    Postgres have to be downloaded. If all goes well it should end with something 
    like:
    ```
    web_1  | [30/Nov/2020 16:46:36] "GET /admin/jsi18n/ HTTP/1.1" 200 3187
    web_1  | Watching for file changes with StatReloader
    web_1  | Performing system checks...
    web_1  | 
    web_1  | System check identified no issues (0 silenced).
    web_1  | December 01, 2020 - 08:49:29
    web_1  | Django version 3.1.3, using settings 'config.settings'
    web_1  | Starting development server at http://0.0.0.0:8100/
    web_1  | Quit the server with CONTROL-C.
    ```

6) Create the database tables

    Open a new command line window for the next steps:
    ```
    docker-compose exec web python manage.py migrate
    ```

7) Create the admin user
    ```
    docker-compose exec web python manage.py createsuperuser
    ```
    Fill in: aesset-rm-admin with password testpass123, or something else ;-)

8) check the app on:
    - http://localhost:8100
    - http://localhost:8100/admin   -> log in with the user you created 

9) Stop the docker images with the application and database:
    
    By pressing CTRL+C in the window where you started the images.

## Start and stop the application
From now on you can start and stop the application with the following 
commands executed in the stripped/ directory: 
```
docker-compose up         # starts the application with log in foreground
docker-compose up -d      # starts the application in the background
docker-compose up --build # rebuilds and starts the application
docker-compose down       # shuts down the application
docker-compose logs -f    # follow the logging
```
 
## Database location
   
The database is placed local on your pc. The location depends on the
installation of Docker, but it is something like:
```
var/lib/docker/volumes/aesset-rm_postgres_data
```

## Upgrade the application

Update the sources: go to the 'stripped' directory on the command line and type:
```
git pull
```

Start the application, with the build option, maybe not needed, but sometimes it is:
```
docker-compose up --build
```

Go to a new commandline window in the 'stripped' directory and type:
```
docker-compose exec web python manage.py migrate
```
This updates the database with the latest structure changes

# Reinitialize the database

Want to start all over? Easy! The database is a 'volume' defined in the 
docker-compose. By removing that volume and restart the containers the
database is recreated. After that you run the migrates and create the 
superuser again:

Check the volumes: 
```
$ docker volume ls
DRIVER    VOLUME NAME
local     aesset-rm_postgres_data      <-- this is the one
```
If you want to get more info about it:
```
$ docker volume inspect aesset<tab> <-- press tab for auto completion
```
Then to remove:
```
$ docker volume rm aesset<tab>
```
Start the containers, create the tables and create superuser:
```
$ docker-compose up -d
$ docker-compose exec web python manage.py migrate
$ docker-compose exec web python manage.py createsuperuser
```
Fill in: aesset-rm-admin with password testpass123, or something else ;-)

And you're back in business!

#### Author
- Eelco Aartsen
- eelco@aesset.nl
- AESSET IT - The Netherlands
- www.aesset.nl




CHANGELOG.md
Changelog for xyz
=================

1.1.3 (unreleased)
------------------

- Nothing changed yet.


1.1.2 (2021-01-24)
------------------

- Testing git commits log II


1.1.1 (2021-01-24)
------------------

- Testing git commits log

- testing prerelase entry point


1.1.0 (2021-01-24)
------------------

- [RM-101](https://aesset.atlassian.net/browse/RM-101) Lekker bezig voor versie 1.1.0


1.0.0 (2021-01-24)
------------------

- [RM-196](https://aesset.atlassian.net/browse/RM-196) En weer wat via addchangelogentry

- [RM-95](https://aesset.atlassian.net/browse/RM-95) Changes correction for 0.1.4

- [RM-95](https://aesset.atlassian.net/browse/RM-95) Changes for 0.1.4

- [RM-195](https://aesset.atlassian.net/browse/RM-195) Gewoon wat met de hand toegevoegd aan de Changelog


0.1.3 (2021-01-24)
------------------

- RE-100 Changed README with some changes

- RE-99 Changed README with some changes


0.1.0 (2020-12-31)
----------------

- Testing123

- Did something

0.0.1(2020-12-24)
----------------

- First install

util.py
import logging
import markdown

from zest.releaser.utils import execute_command

logger = logging.getLogger(__name__)

def convert_changelog_to_html(data):
    logger.info("Convert CHANGELOG.md to CHANGELOG.html")
    with open('CHANGELOG.md', 'r') as f:
        text = f.read()
        html = markdown.markdown(text)


    with open('templates/CHANGELOG.html', 'w') as f:
        f.write("{% extends '_base.html' %}\n")
        f.write("{% load static %}\n")
        f.write("{% block title %}NPO-RM Changelog {% endblock title %}\n")
        f.write("{% block content %}\n")
        f.write("<div class='container-fluid pl-3'>\n")
        f.write("<br>\n")
        f.write(html)
        f.write("\n</div>\n")
        f.write("{% endblock content %}\n")

    logger.info('execute_command(["git", "add", "."]):')
    logger.info(execute_command(["git", "add", "."]))
    logger.info('execute_command(["git", "commit", "-a", "-m", "new CHANGELOG.html"])')
    logger.info(execute_command(["git", "commit", "-a", "-m", "new CHANGELOG.html"]))


config/settings.py
"""
Django settings for config project.

Generated by 'django-admin startproject' using Django 3.1.3.

For more information on this file, see
https://docs.djangoproject.com/en/3.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.1/ref/settings/
"""

from pathlib import Path
import os

with open("config/version.txt") as v_file:
    APP_VERSION_NUMBER = v_file.read()

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent
ENVIRONMENT = 'stripped'

# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.1/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = os.environ.get('SECRET_KEY', default='JustADefaultSecretKey')

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = 'DEBUG'

ALLOWED_HOSTS = ['*',]


# Application definition

INSTALLED_APPS = [
    'whitenoise.runserver_nostatic',   # disable Django’s static file handling and allow WhiteNoise to take over
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.sites',

    # third party
    'crispy_forms',
    'allauth',
    'allauth.account',

    # local apps
    'rm.apps.RmConfig',
    'users.apps.UsersConfig',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'whitenoise.middleware.WhiteNoiseMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]

ROOT_URLCONF = 'config.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [os.path.join(BASE_DIR, 'templates')],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'rm.context_processors.app_version_number'
            ],
        },
    },
]

WSGI_APPLICATION = 'config.wsgi.application'


# Database
# https://docs.djangoproject.com/en/3.1/ref/settings/#databases

# defaults are for local use, as long as we have no environment props there
# db_name = 'POSTGRES_DB'
# db_user = 'POSTGRES_USER'
# db_password = 'POSTGRES_PASSWORD'
# db_host = 'DB_HOST'
#
# DATABASES = {
#     'default': {
#         'ENGINE': 'django.db.backends.postgresql',
#         'NAME': db_name,
#         'USER': db_user,
#         'PASSWORD': db_password,
#         'HOST': db_host,
#         'PORT': '5432',
#     }
# }

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

# Password validation
# https://docs.djangoproject.com/en/3.1/ref/settings/#auth-password-validators

AUTH_USER_MODEL = 'users.CustomUser'

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]

# Static files (CSS, JavaScript, Images)

STATIC_URL = '/static/'
STATICFILES_DIRS = [os.path.join(BASE_DIR, 'static'), ]   # local development dir
STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')
STATICFILES_FINDERS = [
    "django.contrib.staticfiles.finders.FileSystemFinder",      # first in /static/
    "django.contrib.staticfiles.finders.AppDirectoriesFinder"   # then in /<app>/static/
]

# WhiteNoise comes with a storage backend which automatically takes care of
# compressing your files and creating unique names for each version so they
# can safely be cached forever. To use it, just add this to your settings.py:
STATICFILES_STORAGE = 'whitenoise.storage.CompressedManifestStaticFilesStorage'

# If you want to apply compression but don’t want the caching behaviour then
# you can use:
#
# STATICFILES_STORAGE = 'whitenoise.storage.CompressedStaticFilesSto

# Internationalization
# https://docs.djangoproject.com/en/3.1/topics/i18n/

LANGUAGE_CODE = 'nl-nl'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.1/howto/static-files/

STATIC_URL = '/static/'

MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media')

# LOGGING 123456

LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'formatters': {
        'verbose': {
            'format': '{levelname} {asctime} {module} {process:d} {thread:d} {message}',
            'style': '{',
        },
        'simple': {
            'format': '{levelname}:  {message} (Module:{module}.py)',
            'style': '{',
        },
    },
    'handlers': {
        'console': {
            'class': 'logging.StreamHandler',
            'formatter': 'simple'
        },
    },
    'root': {
        'handlers': ['console'],
        'level': 'DEBUG',
    },
}

# production
if ENVIRONMENT == 'prod':
    SECURE_SSL_REDIRECT = True
    SECURE_HSTS_SECONDS = 2592000
    SECURE_HSTS_INCLUDE_SUBDOMAINS = True
    SECURE_HSTS_PRELOAD = True
    SECURE_CONTENT_TYPE_NOSNIFF = True
    SESSION_COOKIE_SECURE = True
    CSRF_COOKIE_SECURE = True
    SECURE_REFERRER_POLICY = 'same-origin'
    SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDER_PROTO', 'https')

# Django allauth config
LOGIN_REDIRECT_URL = 'home'
ACCOUNT_LOGOUT_REDIRECT = 'home'
ACCOUNT_SIGNUP_PASSWORD_ENTER_TWICE = False

SITE_ID = 1

AUTHENTICATION_BACKENDS = [
    # Needed to login by username in Django admin, regardless of `allauth`
    'django.contrib.auth.backends.ModelBackend',

    # `allauth` specific authentication methods, such as login by e-mail
    'allauth.account.auth_backends.AuthenticationBackend',
]

# temporary send email to console
EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'


CRISPY_TEMPLATE_PACK = 'bootstrap4'

config/settings_pycharm.py
"""
Django settings for config project  ---- FOR USE IN PYCHARM

Generated by 'django-admin startproject' using Django 3.1.3.

For more information on this file, see
https://docs.djangoproject.com/en/3.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.1/ref/settings/
"""

from pathlib import Path
import os

with open("config/version.txt") as v_file:
    APP_VERSION_NUMBER = v_file.read()

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent
ENVIRONMENT = 'test-local'

# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.1/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'bm5d)*t)bp95-q*gq85k3-e^=)r)+-8grp+&wu+d#^eji(enx9'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = ['localhost', '127.0.0.1']


# Application definition

INSTALLED_APPS = [
    'whitenoise.runserver_nostatic',   # disable Django’s static file handling and allow WhiteNoise to take over
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.sites',

    # third party
    'allauth',
    'allauth.account',

    # local apps
    'rm.apps.RmConfig',
    'users.apps.UsersConfig',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'whitenoise.middleware.WhiteNoiseMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]

ROOT_URLCONF = 'config.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [os.path.join(BASE_DIR, 'templates')],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'rm.context_processors.app_version_number'
            ],
        },
    },
]

WSGI_APPLICATION = 'config.wsgi.application'


# Database
# https://docs.djangoproject.com/en/3.1/ref/settings/#databases

# defaults are for local use, as long as we have no environment props there
db_name = 'aesset_rm_db'
db_user = 'aesset_rm_user'
db_password = 'testpass123'
db_host = 'localhost'

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql',
        'NAME': db_name,
        'USER': db_user,
        'PASSWORD': db_password,
        'HOST': db_host,
        'PORT': '5432',
    }
}


# Password validation
# https://docs.djangoproject.com/en/3.1/ref/settings/#auth-password-validators

AUTH_USER_MODEL = 'users.CustomUser'

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]

# Static files (CSS, JavaScript, Images)

STATIC_URL = '/static/'
STATICFILES_DIRS = [os.path.join(BASE_DIR, 'static'), ]   # local development dir
STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')
STATICFILES_FINDERS = [
    "django.contrib.staticfiles.finders.FileSystemFinder",      # first in /static/
    "django.contrib.staticfiles.finders.AppDirectoriesFinder"   # then in /<app>/static/
]

# WhiteNoise comes with a storage backend which automatically takes care of
# compressing your files and creating unique names for each version so they
# can safely be cached forever. To use it, just add this to your settings.py:
STATICFILES_STORAGE = 'whitenoise.storage.CompressedManifestStaticFilesStorage'

# If you want to apply compression but don’t want the caching behaviour then
# you can use:
#
# STATICFILES_STORAGE = 'whitenoise.storage.CompressedStaticFilesSto

# Internationalization
# https://docs.djangoproject.com/en/3.1/topics/i18n/

LANGUAGE_CODE = 'nl-nl'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.1/howto/static-files/
#
# settings_pycharm.py
# ===================
# When running tests in pychar, use this file for your 'custom settings' in the
# test run configuration.

STATIC_URL = '/static/'

MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media')

# LOGGING 123456

LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'formatters': {
        'verbose': {
            'format': '{levelname} {asctime} {module} {process:d} {thread:d} {message}',
            'style': '{',
        },
        'simple': {
            'format': '{levelname}:  {message} (Module:{module}.py)',
            'style': '{',
        },
    },
    'handlers': {
        'console': {
            'class': 'logging.StreamHandler',
            'formatter': 'simple'
        },
    },
    'root': {
        'handlers': ['console'],
        'level': 'DEBUG',
    },
}

# production
if ENVIRONMENT == 'prod':
    SECURE_SSL_REDIRECT = True
    SECURE_HSTS_SECONDS = 2592000
    SECURE_HSTS_INCLUDE_SUBDOMAINS = True
    SECURE_HSTS_PRELOAD = True
    SECURE_CONTENT_TYPE_NOSNIFF = True
    SESSION_COOKIE_SECURE = True
    CSRF_COOKIE_SECURE = True
    SECURE_REFERRER_POLICY = 'same-origin'
    SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDER_PROTO', 'https')

# Django allauth
SITE_ID = 1

AUTHENTICATION_BACKENDS = [
    # Needed to login by username in Django admin, regardless of `allauth`
    'django.contrib.auth.backends.ModelBackend',

    # `allauth` specific authentication methods, such as login by e-mail
    'allauth.account.auth_backends.AuthenticationBackend',
]

# temporary send email to console
EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
